gossip/state/state.go

```diff
@@ -432,9 +447,15 @@ func (s *GossipStateProviderImpl) processStateRequests() {
 // handleStateRequest handles state request message, validate batch size, reads current leader state to
 // obtain required blocks, builds response message and send it back
 func (s *GossipStateProviderImpl) handleStateRequest(msg protoext.ReceivedMessage) {
+
+	if !s.CanHandleStateRequest() {
```

Comment:
Is this check required?

Reply:
Yes, only endorser can do this.


In fabric-peer-ext
func (s *gossipStateProviderExtension) CanHandleStateRequest() bool {
	return roles.IsEndorser()
}

Reply:
I'm wondering if we can also create an extension function that performs the default behaviour? I don't really like the CanXXX functions.